subsys/net/lib/wifi_ready/wifi_ready.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/net/net_event.h>
+#include <supp_events.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <net/wifi_ready.h>
+
+LOG_MODULE_REGISTER(wifi_ready, CONFIG_WIFI_READY_EVENT_HANDLING_LOG_LEVEL);
+
+static app_callback_t callback;
+static struct net_mgmt_event_callback net_wpa_supp_cb;
+
+static void handle_wpa_supp_ready(struct net_mgmt_event_callback *cb)
+{
+	if (callback.is_wifi_ready) {
+		callback.is_wifi_ready();
+	}
+}
+
+static void handle_wpa_supp_not_ready(struct net_mgmt_event_callback *cb)
+{
+	LOG_INF("Supplicant is not ready");
+}
```

Comment:
Why 2 different functions (for ready and not ready). Can't it be just 1 function with a bool parameter to specify the readiness?

Same comment for callback. Are 2 different callbacks needed for ready and not ready?